velox/functions/remote/server/RemoteFunctionRestService.cpp

```diff
@@ -0,0 +1,320 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "RemoteFunctionRestService.h"
+
+#include <boost/beast/version.hpp>
+#include <serializers/PrestoSerializer.h>
+#include "velox/expression/Expr.h"
+#include "velox/type/fbhive/HiveTypeParser.h"
+#include "velox/vector/VectorStream.h"
+
+namespace facebook::velox::functions {
+
+namespace {
+
+struct InternalFunctionSignature {
+  std::vector<std::string> argumentTypes;
+  std::string returnType;
+};
+
+std::map<std::string, InternalFunctionSignature> internalFunctionSignatureMap =
+    {
+        {"remote_abs", {{"integer"}, "integer"}},
+        {"remote_plus", {{"bigint", "bigint"}, "bigint"}},
+        {"remote_divide", {{"double", "double"}, "double"}},
+        {"remote_substr", {{"varchar", "integer"}, "varchar"}},
+        // Add more functions here as needed, registerRemoteFunction should be
+        // called to use the functions mentioned in this map
+};
+
+TypePtr deserializeType(const std::string& input) {
```

Comment:
There is a lot of repetition between this code and https://github.com/facebookincubator/velox/blob/main/velox/functions/remote/server/RemoteFunctionService.cpp. Please can you refactor.

Reply:
introduced the RemoteFunctionHelper.h and moved the duplicate code.